h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -499,18 +501,48 @@ public void onModelFailure(Model failedModel, Consumer<Object[]> withFailedModel
          */
         private int[] nextModelIndices() {
           int[] hyperparamIndices =  new int[_hyperParamNames.length];
-
+          Set<String> gam_specific_params = new HashSet<>(Arrays.asList("bs", "scale", "num_knots"));
```

Comment:
should be a private static final constant with unmodifiableSet

Reply:
@honzasterba:

Basically, this is what is going on with GAM:

1. a user can choose which sets of gam columns to be included to transform into gam columns.  Some times you may want to run gam with say [col0, col1] and sometimes, you may want to try to run gam with [col1, col2, col3].  We can make the gridsearch more restrictive for the users to force them to only choose a subset of 2 gam columns or N-gam columns.  Doing it this way, all the scale/num_knots/bs will have the same length.

2. Our current implementation is to allow users to choose any gam column sizes.  Then, in scale/num_knots/bs, users can also add different sizes.  However, our code will automatically choose the correct size scale/num_knots/bs according to the gam_column that is randomly chosen.

If we go with 1., we need to make sure the documentation reflect that the user can gridsearch over gam columns of a fixed size.  If they want to gridsearch over gam columns of different sizes, they will need to do it manually.  Let me know what you think.  Thanks, Wendy

Reply:
so we could also fix this by mandating that all scale/... arrays be same or bigger than smallest GAM-columns array

Reply:
@honzasterba a non-algo specific is used.